editor/xml_feature.cpp

```diff
@@ -611,6 +776,36 @@ XMLFeature ToXML(osm::EditableMapObject const & object, bool serializeType)
   return toFeature;
 }
 
+XMLFeature TypeToXML(uint32_t type, feature::GeomType geomType, m2::PointD mercator)
+{
+  ASSERT(geomType == feature::GeomType::Point, ("Only point features can be added"));
+  XMLFeature toFeature(XMLFeature::Type::Node);
+  toFeature.SetCenter(mercator);
+
+  // Set Type
+  if (ftypes::IsRecyclingCentreChecker::Instance()(type))
+  {
+    toFeature.SetTagValue("amenity", "recycling");
+    toFeature.SetTagValue("recycling_type", "centre");
+  }
+  else if (ftypes::IsRecyclingContainerChecker::Instance()(type))
+  {
+    toFeature.SetTagValue("amenity", "recycling");
+    toFeature.SetTagValue("recycling_type", "container");
+  }
+  else
+  {
+    string const strType = classif().GetReadableObjectName(uint32_t(type));
+    strings::SimpleTokenizer iter(strType, "-");
+    string_view const k = *iter;
+
+    CHECK(++iter, ("Processing Type failed: ", strType));
+    // Main type is always stored as "k=amenity v=restaurant".
+    toFeature.SetTagValue(k, *iter);
```

Comment:
I suspect that we have more 3-arity types besides amenity-recycling-xxx
Add ASSERT/CHECK(!(++iter), ("Add an exception:", strType))

Reply:
:+1: Added an assertion